backend/internal/app/parkserver/server.go

```diff
@@ -6,35 +6,63 @@ import (
 	"net"
 	"net/http"
 
+	authRepo "github.com/ParkWithEase/parkeasy/backend/internal/pkg/repositories/auth"
+	userRepo "github.com/ParkWithEase/parkeasy/backend/internal/pkg/repositories/user"
 	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/routes"
-	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/services"
+	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/services/auth"
+	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/services/user"
+	"github.com/alexedwards/scs/v2"
 	"github.com/danielgtaylor/huma/v2"
 	"github.com/danielgtaylor/huma/v2/adapters/humago"
 )
 
+type Config struct {
+	// The address to run the server on
+	Addr string
+	// Whether to run server in insecure mode. This allows cookies to be transferred over plain HTTP.
+	Insecure bool
+}
+
 // Register all routes
-func RegisterRoutes(api huma.API) {
-	huma.AutoRegister(api, routes.NewGreetingRoute(&services.SimpleGreeting{}))
+func RegisterRoutes(api huma.API, sessionManager *scs.SessionManager) {
+	authMiddleware := routes.NewSessionMiddleware(api, sessionManager)
+	api.UseMiddleware(authMiddleware)
+
+	authRepo := authRepo.NewMemoryRepository()
+	authService := auth.NewService(authRepo)
+	authRoute := routes.NewAuthRoute(authService, sessionManager)
+
+	userRepo := userRepo.NewMemoryRepository()
+	userService := user.NewService(*authService, userRepo)
```

Comment:
Just wondering why is authRepo return as a variable but authService is returned as a pointer to the service.